modules/web_hosting/main.tf

```diff
@@ -0,0 +1,550 @@
+# Service Account to connect to Cloud SQL
+#########################################################################
+
+resource "google_service_account" "sa_p_cloud_sql" {
+  project  = local.project.project_id
+  account_id   = "gce-sql-sa"
+  display_name = "Service Account to connect Cloud SQL"
+}
+
+#########################################################################
+# Startup script for VMs in vpc-xlb
+#########################################################################
+
+data "template_file" "sample_webapp_metadata_startup_script" {
+    template = "${file("${path.module}/scripts/build/startup_scripts/sample_app/sample_webapp.sh.tpl")}"
+    vars = {
+        INSTANCE_CONNECTION_NAME = module.sql_db_postgresql.instance_connection_name
+        CLOUD_SQL_DATABASE_NAME  = "postgres"
+        CLOUD_SQL_USERNAME       = module.sql_db_postgresql.additional_users[0].name
+        CLOUD_SQL_PASSWORD       = module.sql_db_postgresql.additional_users[0].password
+
+    }
+}
+
+data "template_file" "sample_app_metadata_startup_script" {
+    template = "${file("${path.module}/scripts/build/startup_scripts/sample_app/sample_app.sh.tpl")}"
+}
+
+#########################################################################
+# Creating GCE VMs in vpc-xlb
+#########################################################################
+
+
+resource "google_compute_instance" "web1_vpc_xlb" {
+  project      = local.project.project_id
+  zone         = "${var.region}-a"
+  name         = "web1-vpc-xlb"
+  machine_type = "f1-micro"
+  allow_stopping_for_update = true
+  metadata_startup_script   = data.template_file.sample_webapp_metadata_startup_script.rendered
+  metadata = {
+    enable-oslogin = true
+  }
+  boot_disk {
+    initialize_params {
+      image = "debian-11-bullseye-v20220822"
+    }
+  }
+ 
+  network_interface {
+    subnetwork         = google_compute_subnetwork.subnetwork_primary.name
+    subnetwork_project = local.project.project_id
+    network_ip         = cidrhost(google_compute_subnetwork.subnetwork_primary.ip_cidr_range, 2)
+    # access_config {
+    #   // Ephemeral IP
+    # }
+  }
+  service_account {
+    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
+    email  = google_service_account.sa_p_cloud_sql.email
+    scopes = ["cloud-platform"]
+  }
+ 
+  depends_on = [
+    time_sleep.wait_120_seconds,
+    google_compute_router_nat.nat_gw_region_primary
+    ]
+}
+
+resource "google_compute_instance" "web2_vpc_xlb" {
+  project      = local.project.project_id
+  zone         = "${var.region}-b"
+  name         = "web2-vpc-xlb"
+  machine_type = "f1-micro"
+  allow_stopping_for_update = true
+  metadata_startup_script   = data.template_file.sample_webapp_metadata_startup_script.rendered
+  metadata = {
+    enable-oslogin = true
+  }
+  boot_disk {
+    initialize_params {
+      image = "debian-11-bullseye-v20220822"
```

Comment:
Can we replace this with a data-block?

```
data "google_compute_image" "debian_11_bullseye" {
  family = "debian-11"
  project = "debian-cloud"
}

...

boot_disk {
  initialize_params {
    image = data.google_compute_image.debian_11_bullseye.self_link
  }
}
```

Perhaps we can create a variable for this as well, so users can pass in their own image if they want to use a different one.  Problem is that I don't know how this will impact the remainder of the flow, in terms of startup scripts etc.

Reply:
Created the data block. Done